canopy_bioemi_mod.F90

```diff
@@ -0,0 +1,310 @@
+                CTEMP_SUN(i)   = interp_linear1_internal((/ FCH*(4.0_rk/5.0_rk),FCH /), &
+                    (/ CTEMP_2_SUN,CTEMP_1_SUN /),ZK(i))
+                DTEMP_SUN(i)   = interp_linear1_internal((/ FCH*(4.0_rk/5.0_rk),FCH /), &
+                    (/ DTEMP_2_SUN,DTEMP_1_SUN /),ZK(i))
+                CTEMP_SHADE(i) = interp_linear1_internal((/ FCH*(4.0_rk/5.0_rk),FCH /), &
+                    (/ CTEMP_2_SHADE,CTEMP_1_SHADE /),ZK(i))
+                DTEMP_SHADE(i) = interp_linear1_internal((/ FCH*(4.0_rk/5.0_rk),FCH /), &
+                    (/ DTEMP_2_SHADE,DTEMP_1_SHADE /),ZK(i))
+            else if (ZK(i) .le. FCH*(4.0_rk/5.0_rk) .and. ZK(i) .gt. FCH*(3.0_rk/5.0_rk)) then  !Level 2 - 3
+                CTEMP_SUN(i)   = interp_linear1_internal((/ FCH*(3.0_rk/5.0_rk),FCH*(4.0_rk/5.0_rk) /), &
+                    (/ CTEMP_3_SUN,CTEMP_2_SUN /),ZK(i))
+                DTEMP_SUN(i)   = interp_linear1_internal((/ FCH*(3.0_rk/5.0_rk),FCH*(4.0_rk/5.0_rk) /), &
+                    (/ DTEMP_3_SUN,DTEMP_2_SUN /),ZK(i))
+                CTEMP_SHADE(i) = interp_linear1_internal((/ FCH*(3.0_rk/5.0_rk),FCH*(4.0_rk/5.0_rk) /), &
+                    (/ CTEMP_3_SHADE,CTEMP_2_SHADE /),ZK(i))
+                DTEMP_SHADE(i) = interp_linear1_internal((/ FCH*(3.0_rk/5.0_rk),FCH*(4.0_rk/5.0_rk) /), &
+                    (/ DTEMP_3_SHADE,DTEMP_2_SHADE /),ZK(i))
+            else if (ZK(i) .le. FCH*(3.0_rk/5.0_rk) .and. ZK(i) .gt. FCH*(2.0_rk/5.0_rk)) then  !Level 3 - 4
+                CTEMP_SUN(i)   = interp_linear1_internal((/ FCH*(2.0_rk/5.0_rk),FCH*(3.0_rk/5.0_rk) /), &
+                    (/ CTEMP_4_SUN,CTEMP_3_SUN /),ZK(i))
+                DTEMP_SUN(i)   = interp_linear1_internal((/ FCH*(2.0_rk/5.0_rk),FCH*(3.0_rk/5.0_rk) /), &
+                    (/ DTEMP_4_SUN,DTEMP_3_SUN /),ZK(i))
+                CTEMP_SHADE(i) = interp_linear1_internal((/ FCH*(2.0_rk/5.0_rk),FCH*(3.0_rk/5.0_rk) /), &
+                    (/ CTEMP_4_SHADE,CTEMP_3_SHADE /),ZK(i))
+                DTEMP_SHADE(i) = interp_linear1_internal((/ FCH*(2.0_rk/5.0_rk),FCH*(3.0_rk/5.0_rk) /), &
+                    (/ DTEMP_4_SHADE,DTEMP_3_SHADE /),ZK(i))
+            else if (ZK(i) .le. FCH*(2.0_rk/5.0_rk) ) then  !Level 4 - Bottom
+                CTEMP_SUN(i)   = interp_linear1_internal((/ ZK(1),FCH*(2.0_rk/5.0_rk) /), &
+                    (/ CTEMP_5_SUN,CTEMP_4_SUN /),ZK(i))
+                DTEMP_SUN(i)   = interp_linear1_internal((/ ZK(1),FCH*(2.0_rk/5.0_rk) /), &
+                    (/ DTEMP_5_SUN,DTEMP_4_SUN /),ZK(i))
+                CTEMP_SHADE(i) = interp_linear1_internal((/ ZK(1),FCH*(2.0_rk/5.0_rk) /), &
+                    (/ CTEMP_5_SHADE,CTEMP_4_SHADE /),ZK(i))
+                DTEMP_SHADE(i) = interp_linear1_internal((/ ZK(1),FCH*(2.0_rk/5.0_rk) /), &
+                    (/ DTEMP_5_SHADE,DTEMP_4_SHADE /),ZK(i))
+            end if
+        end do
+
+        PPFD_SUN     = FRAC_PPFD * SFCRAD * EXP(CTEMP_SUN + DTEMP_SUN * LAI)  !W/m2 --> umol m-2 s-1
```

Comment:
0.5 does not do the conversion to umol m-2 s-1. That's what the 4.5 was doing. 1 W m-2 ≈ 4.6 umol m-2 s-1 for the visible waveband for typical clear sky conditions.

Reply:
@zmoon   Yes, I understand.  The FRAC_PPFD is 50% the incoming total solar radiation, as in above reference.  

The conversion seems to come the exponential relationship in equation (2) in Silva et al.

![image](https://user-images.githubusercontent.com/26631222/224440806-6dbf4347-f27e-460a-9fc2-58902fe206a5.png)

Reply:
Yeah does seem to suggest that the coeffs somehow include the conversion. I may briefly try it out.

Reply:
<details>
<summary>Seems to check out</summary>

![silva-coeffs_fig1](https://user-images.githubusercontent.com/15079414/224449684-6b8fee5d-27ed-4905-9032-fca50dee9680.png)
</details>

Reply:
@zmoon Yes I did an independent check and that is what I saw too.  I think it looks good, but I am open to using other methods to get at the fraction of sunlit/shaded leaves through the canopy later.

Reply:
I am going to do some more checks, e.g. comparing to my own calculation. On issue is that the LAD profile it is based on is fixed, and most likely not consistent with canopy-app's. But we could improve that later.

Reply:
@zmoon I agree this is an issue with fixing the Tleaf and PPFDleaf on these linear and exponential relationships from Silva et al.  Looking forward to what you find out to help improve this.

Reply:
<details><summary>A little example</summary>

![image](https://user-images.githubusercontent.com/15079414/229199224-4486a0e6-6704-490e-b09f-4f43521be089.png)
</details>

Similar enough, though a little suspicious that the Silva profiles don't get very close to 4.6 (should be at L=0 for clear sky conditions). But let's leave it for now.